### PR TITLE
Filter page

### PR DESCRIPTION
**Pull request by davidwobido**
## ✅ Done
• Add filter page which displays all genre tags
• Activate and deactivate the genres
• Status of selection is lifted up to page instead of button

## 🚧 TBD on new branches
• Filter button + filter function tbd on new branch: [See issue here](https://github.com/davidwobido/festival-filter/issues/14)
• Final styling tbd on new branch: [See issue here](https://github.com/davidwobido/festival-filter/issues/36)

## Screenshot
image

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -2,6 +2,7 @@
   "name": "template",
   "version": "0.0.0",
   "scripts": {
+    "dev": "concurrently \"npm run dev-client\" \"npm run dev-server\" \"npm run storybook\"",
```
Comment (lmachens) [inline package.json:5]: This change is unrelated to `Filter page`. I would create a new pull request with this change only.

File: src/pages/Filter/FilterPage.tsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    { text: "Pop", selected: false, id: 0 },
+    { text: "HipHop", selected: false, id: 1 },
+    { text: "Rock", selected: false, id: 2 },
+    { text: "Indie", selected: false, id: 3 },
+    { text: "Punk", selected: false, id: 4 },
+    { text: "Metal", selected: false, id: 5 },
+    { text: "Electronic", selected: false, id: 6 },
+    { text: "Reggae", selected: false, id: 7 },
+    { text: "Jazz", selected: false, id: 8 },
+    { text: "Classical", selected: false, id: 9 },
```
Comment (F-Kirchhoff) [inline src/pages/Filter/FilterPage.tsx:17]: this looks redundant, maybe generate them from an array with a map?

File: src/components/GenreTag/GenreTag.tsx [hunk 3]
```diff
@@ -2,11 +2,23 @@ import styles from "./GenreTag.module.css";
...
+  };
+  handleClick: (value: number) => void;
 };
 
-function GenreTag({ text }: GenreTagProps): JSX.Element {
-  return <button className={styles.tag}>{text}</button>;
+function GenreTag({ tag, handleClick }: GenreTagProps): JSX.Element {
+  return (
+    <button
+      className={tag.selected ? styles.tag_selected : styles.tag_unselected}
```
Comment (lmachens) [inline src/components/GenreTag/GenreTag.tsx:16]: If you like BEM, it would look like this:

```suggestion
      className={`${styles.tag} ${tag.selected ? styles.tag_selected : styles.tag_unselected}`}
```

File: src/pages/Filter/FilterPage.module.css [hunk 4]
```diff
@@ -0,0 +1,22 @@
+.wrapper {
+  display: grid;
+  height: 100vh;
```
Comment (lmachens) [inline src/pages/Filter/FilterPage.module.css:3]: Instead of adding `100vh` to each page, you could add this once in your App

File: src/pages/Filter/FilterPage.module.css [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+  justify-items: center;
+}
+
+.content {
+  padding: 1.25rem;
+}
+
+.preline {
+  composes: text-preline from global;
+}
```
Comment (lmachens) [inline src/pages/Filter/FilterPage.module.css:15]: You could use `text-preline` directly, because `preline` is equal`.

File: src/pages/Filter/FilterPage.tsx [hunk 6]
```diff
@@ -0,0 +1,51 @@
+import styles from "./FilterPage.module.css";
+import GenreTag from "../../components/GenreTag/GenreTag";
+import Button from "../../components/Buttons/Button";
+import { useState } from "react";
+
+function SelectGenre() {
+  const [tags, setTags] = useState([
```
Comment (lmachens) [inline src/pages/Filter/FilterPage.tsx:7]: I would explicit set the type: 

```suggestion
  const [tags, setTags] = useState<Tag[]>([
```

https://github.com/typescript-cheatsheets/react/blob/main/README.md#basic-cheatsheet-table-of-contents

File: src/pages/Filter/FilterPage.tsx [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+    const newTags = [...tags];
+    const tag = newTags.find((tag) => tag.id === id);
+    if (tag) {
+      tag.selected = !tag.selected;
+    }
+    setTags(newTags);
+  }
+
+  return (
+    <>
```
Comment (lmachens) [inline src/pages/Filter/FilterPage.tsx:30]:
```suggestion
```

File: src/pages/Filter/FilterPage.tsx [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+    { text: "HipHop", selected: false, id: 1 },
+    { text: "Rock", selected: false, id: 2 },
+    { text: "Indie", selected: false, id: 3 },
+    { text: "Punk", selected: false, id: 4 },
+    { text: "Metal", selected: false, id: 5 },
+    { text: "Electronic", selected: false, id: 6 },
+    { text: "Reggae", selected: false, id: 7 },
+    { text: "Jazz", selected: false, id: 8 },
+    { text: "Classical", selected: false, id: 9 },
+  ]);
```
Comment (lmachens) [inline src/pages/Filter/FilterPage.tsx:18]:
```suggestion
const initialTags = [
  { text: "Pop", selected: false, id: 0 },
  { text: "HipHop", selected: false, id: 1 },
  { text: "Rock", selected: false, id: 2 },
  { text: "Indie", selected: false, id: 3 },
  { text: "Punk", selected: false, id: 4 },
  { text: "Metal", selected: false, id: 5 },
  { text: "Electronic", selected: false, id: 6 },
  { text: "Reggae", selected: false, id: 7 },
  { text: "Jazz", selected: false, id: 8 },
  { text: "Classical", selected: false, id: 9 },
];

function SelectGenre() {
  const [tags, setTags] = useState(initialTags);
```

or 

```suggestion
const initialTags = ["Pop", "HipHop", "Rock", "Indie"].map((genre, index) => ({
  text: genre,
  selected: false
  id: index
});

function SelectGenre() {
  const [tags, setTags] = useState(initialTags);
```

**Review by lmachens — COMMENTED** [review]
o(*^＠^*)o

- `FF_Animation_loading_rz_pink.json` is a long name and I don't know what "FF" or "rz" is standing for. Maybe rename it to `loading.json`

File: src/pages/Filter/FilterPage.module.css (hunk 4 above)
Comment (davidwobido) [reply src/pages/Filter/FilterPage.module.css:3]: Will do this on a new branch

File: src/components/GenreTag/GenreTag.tsx (hunk 3 above)
Comment (davidwobido) [reply src/components/GenreTag/GenreTag.tsx]: This is great! thanks!
